source/slang/ir-legalize-types.cpp

```diff
@@ -1445,6 +1607,62 @@ static LegalVal legalizeGlobalConstant(
     }
 }
 
+static LegalVal legalizeGlobalParam(
+    IRTypeLegalizationContext*  context,
+    IRGlobalParam*              irGlobalParam)
+{
+    // Legalize the type for the variable's value
+    auto legalValueType = legalizeType(
+        context,
+        irGlobalParam->getFullType());
+
+    RefPtr<VarLayout> varLayout = findVarLayout(irGlobalParam);
+    RefPtr<TypeLayout> typeLayout = varLayout ? varLayout->typeLayout : nullptr;
+
+    switch (legalValueType.flavor)
+    {
+    case LegalType::Flavor::simple:
```

Comment:
Two pedantic points... LegalVal::makeSimple etc, might be a little clearer and flavor enum class values been upper camel.

Reply:
Yeah, the case names here were from a brief period where I tried to get (over)zealous about the idea that "all values are `lowerCamelCase`." It turns out to be kind of a mess in C++, though, so I abandoned it shortly after. (The fact that it looks weird to eyes accustomed to C/C++ was kind of annoying, but I figured I'd get used to it. The sticking point was that I never realized how often you need/want an `enum` case name that would overlap with a keyword. E.g., `SomeMode::Default` is a really common case)

As you know, I'm not into cleaning up formatting/style stuff that is unrelated to the functional change being made, but if/when we do a code freeze and spend a sprint just on style issues this would be something we'd clean up.

On the constructor stuff... yeah, using `make` consistently for value factory functions and `create` consistently for object/pointer factories would probably be a good naming guideline to follow.